client/app/collections/filtered.coffee

```diff
@@ -4,12 +4,22 @@ ContactViewModel = require 'views/models/contact'
 
 tagRegExp = search.pattern 'tag'
 app       = null
-scores    = new WeakMap()
+scores    = new Map()
+
+
+Math.median = (array) ->
+    sorted = array.sort (a, b) -> a - b
+    len = array.length
+
+    if len % 2
+        sorted[Math.floor len / 2]
+    else
+        (sorted[len / 2] + sorted[len / 2 - 1]) / 2
```

Comment:
Not sure about this one: if array has an even length, the median value is the sum of the to middle values divided by 2 (https://en.wikipedia.org/wiki/Median). Is it what you think about?